ClientCore/JSONDataSourceManager.cs

```diff
@@ -0,0 +1,339 @@
+{
+    private static JSONDataSourceManager _instance;
+    public static JSONDataSourceManager Instance => _instance ??= new JSONDataSourceManager();
+
+    private readonly Dictionary<string, DataSource> _dataSources = new Dictionary<string, DataSource>();
+
+    private JSONDataSourceManager() { }
+
+    /// <summary>
+    /// Loads data sources from an INI file's [DataSources] section.
+    /// </summary>
+    /// <param name="iniFile">The INI file to read from</param>
+    public void LoadFromINI(IniFile iniFile)
+    {
+        const string sectionName = "DataSources";
+        var section = iniFile.GetSection(sectionName);
+
+        foreach (var key in section.Keys)
+        {
+            string value = section.GetStringValue(key.Key, string.Empty);
+            if (string.IsNullOrEmpty(value))
+                continue;
+
+            // format: URL,RefreshIntervalSeconds,TimeoutSeconds,Format
+            string[] parts = value.Split(',');
+            if (parts.Length < 1)
+                continue;
+
+            string url = parts[0].Trim();
+            int refreshInterval = parts.Length > 1 ? Conversions.IntFromString(parts[1].Trim(), 300) : 300;
+            int timeout = parts.Length > 2 ? Conversions.IntFromString(parts[2].Trim(), 10) : 10;
+            string format = parts.Length > 3 ? parts[3].Trim().ToLowerInvariant() : "json";
+
+            if (string.IsNullOrEmpty(url) || _dataSources.ContainsKey(key.Key))
+                continue;
+
+            var dataSource = new DataSource(key.Key, url, refreshInterval, timeout, format);
+            _dataSources.Add(key.Key, dataSource);
+        }
+    }
+
+    /// <summary>
+    /// Subscribes to a data source and receives JSON updates.
+    /// </summary>
+    /// <param name="dataSourceId">The ID of the data source</param>
+    /// <param name="callback">Callback invoked when new data is available (json, isError)</param>
+    /// <returns>True if subscription was successful, false if data source not found</returns>
+    public bool Subscribe(string dataSourceId, Action<string, bool> callback)
+    {
+        if (string.IsNullOrEmpty(dataSourceId))
+            return false;
+
+        if (!_dataSources.TryGetValue(dataSourceId, out var dataSource))
+            return false;
+
+        dataSource.Subscribe(callback);
+        return true;
+    }
+
+    /// <summary>
+    /// Unsubscribes from a data source.
+    /// </summary>
+    public void Unsubscribe(string dataSourceId, Action<string, bool> callback)
+    {
+        if (string.IsNullOrEmpty(dataSourceId))
+            return;
+
+        if (_dataSources.TryGetValue(dataSourceId, out var dataSource))
+            dataSource.Unsubscribe(callback);
+    }
+
+    private class DataSource
+    {
+        private readonly string _id;
+        private readonly string _url;
+        private readonly int _refreshIntervalSeconds;
+        private readonly int _timeoutSeconds;
+        private readonly string _format;
```

Comment:
I think this `_format` should be either an enum and/or maybe even a tiny simple interface/class with one method that converts input value (stream?) and gives JSON for arbitrary data representation.